[훈딩] 3주차 뉴스 스탠드

hoongding opened the pull request:
체크 리스트
- [x] 구독 - 그리드 영역 구현
- [x] 구독 - 리스트 영역 구현
- [x] 리스트 영역 애니메이션 구현
- [ ] 코드 리팩토링
- [ ] 테스트 코드 작성

hoongding committed:
💄 style : 리스트 뷰에서 sub news 길이 제한 두고 ... 으로 넘치면 바꾸기

hoongding committed:
✨ feat: 탭 누르면 파란색으로 바꾸기 성공

hoongding committed:
💄 style : 페이지 정보 넣기 위해 css 변경

hoongding committed:
✨ feat: 각 페이지 카운트하기 위해 reducer 변경

hoongding committed:
💄 style : 서브 뉴스 hover 시 underline 안되던 현상 해결

hoongding committed:
✨ feat: 리스트 페이지 nav 바에 현재 페이지 및 전체 페이지 상태 렌더 성공

hoongding committed:
✨ feat: 리스트에서 구독, 해지 완료

hoongding committed:
💄 style : 구독한 언론사 없을때 empty 뷰 완성

hoongding committed:
✨ feat: 구독한 언론사 그리드 완료

hoongding committed:
✨ feat: 내가구독한, 전체 누르면 글씨 색 바꾸기 성공

hoongding committed:
✨ feat: 리스트 뷰에서 구독한 언론 구독버튼, 구독해지 버튼 나누기 성공

hoongding committed:
✨ feat: 다른 뷰로 넘어갔다가 다시 리스트 뷰로 갈때 state를 reset시키기 완료

hoongding committed:
✨ feat: 마이 리스트 뷰  초기 설정

hoongding committed:
💄 style: 내가 구독한 언론사 리스트 뷰 nav 바 overflow 해결 완료

hoongding committed:
✨ feat: 내가 구독한 언론사 리스트 뷰 초기 focus 적용 위해 DOM 변경

hoongding committed:
✨ feat: 헤더의 각 버튼 클릭시 default 값 설정

hoongding committed:
✨ feat: 내가 구독한 언론사 리스트 페이지 탭 클릭 이벤트 구현

hoongding committed:
✨ feat: 내가 구독한 언론 리스트 영역 왼쪽 오른쪽 버튼 클릭 이벤트 구현

hoongding committed:
✨ feat: 각 뷰에서의 왼쪽 오른쪽 버튼 각 상태에 따라 렌더하기 완료!!!!!!!!

hoongding committed:
✨ feat: 스크롤바 현재 선택된 놈 보이게끔 완료

bread1022 commented on js/actions/ActionTypes.js:
```diff
@@ -22,9 +22,14 @@ export const displayActions = {
   HEADER_ALL_MEDIA_BUTTON_CLICK: 'HEADER_ALL_MEDIA_BUTTON_CLICK',
   HEADER_MY_MEDIA_BUTTON_CLICK: 'HEADER_MY_MEDIA_BUTTON_CLICK',
+
+  MINE_LIST_TAB_BUTTON_CLICK: 'MINE_LIST_TAB_BUTTON_CLICK',
+  MINE_LIST_RIGHT_BUTTON_CLICK: 'MINE_LIST_RIGHT_BUTTON_CLICK',
+  MINE_LIST_LEFT_BUTTON_CLICK: 'MINE_LIST_LEFT_BUTTON_CLICK',
```
key와 value의 형태(값(?))가 같은데 따로 객체에 넣어두신 이유가 있으신가요 ?

bread1022 commented on js/components/main/MainCommon/mainButtonEventHandlers.js:
```diff
@@ -76,6 +85,105 @@ export const headerViewChangeBtnClickEventHandler = ({ target }) => {
+) => {
+  const allOrMine = viewOptionData.allOrMine;
+  const gridOrList = viewOptionData.gridOrList;
+  switch (true) {
+    case allOrMine === 'all' && gridOrList === 'grid':
+      gridButtonClickHandler('left', $mainSection, $targetButton);
+      break;
+    case allOrMine === 'all' && gridOrList === 'list':
+      dispatch(displayActionCreator.listLeftBtnClick());
+      break;
+    case allOrMine === 'mine' && gridOrList === 'grid':
+      mineGridButtonClickHandler('left', $mainSection, $buttons);
+      break;
+    case allOrMine === 'mine' && gridOrList === 'list':
+      dispatch(displayActionCreator.mineListLeftBtnClick());
+      break;
```
조건식이 true일때만 실행되는 로직인가요 ? 한 수 배워갑니다.. 👍🏻

bread1022 commented on js/components/main/MainCommon/mainButtonEventHandlers.js:
```diff
@@ -76,6 +85,105 @@ export const headerViewChangeBtnClickEventHandler = ({ target }) => {
+  const targetPageNum = currentPageNum + directionNum;
+  const $targetPage = $pages[targetPageNum];
+  $currentPage.classList.replace('grid', 'none');
+  $targetPage.classList.replace('none', 'grid');
```
wow replace 메서드 활용을 잘하셨네요

bread1022 commented on js/components/main/MainMineList/mainMineList.js:
```diff
@@ -0,0 +1,106 @@
+  // current-category가 있는 li 요소의 위치를 계산하여 스크롤 이동
+  if (currentCategory) {
+    const navItems = document.querySelector('.main-list__nav');
+    const navItemWidth = navItems.offsetWidth;
+    const currentItemWidth = currentCategory.offsetWidth;
+    const currentItemLeft = currentCategory.offsetLeft;
```
offsetWidth, offsetLeft 사용법이 어떻게 되나요 ?

bread1022 commented on js/App.js:
```diff
@@ -20,6 +22,8 @@ const App = () => {
+  MineGrid($main);
+  MineList($main);
 
   root.appendChild(documentFragment);
```
entrypoint 가 매우 깔끔하네요

hoongding committed:
✨ feat: requestAnimationFrame 여러번 실행되던 버그 해결

먼저 페이지 이벤트가 발생하면 무조건 animation을 종료해야함.

그래서 animation start, end action을 만들어 dispatch를 해서 해결했음.

hoongding committed:
✨ feat: Progress Bar 애니메이션까지 구현 완료
